### PR TITLE
Replaced most static components with non-static ones

### LINKED ISSUE #73: Replace all unnecessary static and public fields with non-static and private ones
This is for a better encapsulation.

### PR DESCRIPTION
**Pull request by CraftedMods**
Closes #73. A structural change I wanted to do for a long time. Note that the way the configuration properties are handled is a temporary one, I'll improve that with #33.

### CODE REVIEW

File: src/main/java/pvpmode/internal/server/PvPServerEventHandler.java [hunk 1]
```diff
@@ -136,10 +141,11 @@ public void onLivingHurt (LivingHurtEvent event)
         if (attacker == null || victim == null)
             return;
 
-        if (ServerProxy.activatedPvPLoggingHandlers.size () > 0
+        if (server.getActivatedPvPLoggingHandlers ().size () > 0
             && !MinecraftForge.EVENT_BUS.post (new OnPvPLogEvent (attacker, victim, event.ammount, event.source)))
         {
-            ServerProxy.combatLogManager.log (attacker, victim, event.ammount, event.source);
```
Comment (Mahtaran) [inline src/main/java/pvpmode/internal/server/PvPServerEventHandler.java]: Wait why not use rhe `server` variable here?

File: src/main/java/pvpmode/internal/server/command/PvPCommandList.java [hunk 2]
```diff
@@ -112,7 +115,7 @@ public void processCommand (ICommandSender sender, String[] args)
          * command can determine which entries are the most relevant for the calling
          * player and have to be displayed on the top of the list.
          */
-        for (Object o : ServerProxy.cfg.playerEntityList)
+        for (Object o : PvPMode.instance.getServerProxy ().getServerConfigurationManager ().playerEntityList)
```
Comment (Mahtaran) [inline src/main/java/pvpmode/internal/server/command/PvPCommandList.java]: Again

File: src/main/java/pvpmode/internal/server/command/PvPCommandList.java [hunk 3]
```diff
@@ -211,7 +214,8 @@ public void processCommand (ICommandSender sender, String[] args)
             }
         }
 
-        int playerCountWithoutSender = ServerProxy.cfg.playerEntityList.size () - 1;
+        int playerCountWithoutSender = PvPMode.instance.getServerProxy ()
```
Comment (Mahtaran) [inline src/main/java/pvpmode/internal/server/command/PvPCommandList.java]: And again

File: src/main/java/pvpmode/internal/server/utils/PvPServerUtilsProvider.java [hunk 4]
```diff
@@ -15,19 +16,28 @@
...
+    public PvPServerUtilsProvider (ServerProxy server)
+    {
+        this.server = server;
+    }
+
     @Override
     public EntityPlayerMP getPlayer (String name)
     {
-        return ServerProxy.cfg.func_152612_a (name);
+        return PvPMode.instance.getServerProxy ().getServerConfigurationManager ().func_152612_a (name);
```
Comment (Mahtaran) [inline src/main/java/pvpmode/internal/server/utils/PvPServerUtilsProvider.java]: Again

File: src/main/java/pvpmode/internal/server/utils/PvPServerUtilsProvider.java [hunk 5]
```diff
@@ -15,19 +16,28 @@
...
     }
 
     @Override
     public boolean isOpped (ICommandSender sender)
     {
         if (sender instanceof EntityPlayerMP)
-            return ServerProxy.cfg.func_152596_g ( ((EntityPlayerMP) sender).getGameProfile ());
+            return PvPMode.instance.getServerProxy ().getServerConfigurationManager ()
```
Comment (Mahtaran) [inline src/main/java/pvpmode/internal/server/utils/PvPServerUtilsProvider.java]: Again

File: src/main/java/pvpmode/internal/server/utils/ServerChatUtilsProvider.java [hunk 6]
```diff
@@ -43,10 +51,11 @@ public void postGlobalChatMessages (EnumChatFormatting color, String... messages
             for (String line : message.split ("\n"))
             {
                 ChatComponentText prefix = new ChatComponentText (
-                    ServerProxy.prefixGlobalMessages ? ServerProxy.globalMessagePrefix : "");
+                    server.isPrefixGlobalMessages () ? server.getGlobalMessagePrefix () : "");
                 ChatComponentText text = new ChatComponentText (line);
                 text.getChatStyle ().setColor (color);
-                ServerProxy.cfg.sendChatMsg (prefix.appendSibling (text));
+                PvPMode.instance.getServerProxy ().getServerConfigurationManager ()
```
Comment (Mahtaran) [inline src/main/java/pvpmode/internal/server/utils/ServerChatUtilsProvider.java]: And again

File: src/main/java/pvpmode/internal/server/PvPServerEventHandler.java (hunk 1 above)
Comment (CraftedMods) [reply src/main/java/pvpmode/internal/server/PvPServerEventHandler.java]: The red lines are removed code.

**Comment by CraftedMods** [comment]
You're viewing the changes in a comparison view: Green things were added by the commit, red things were removed.

**Comment by CraftedMods** [comment]
Yeah, you're right. After I did change the configuration properties, I added the fiels, so I don't have to retrieve the proxy every call. The code parts updated before that did still use the long way to the proxy, and I forgot to update them. Thanks for pointing that out!

**Pull request merged**